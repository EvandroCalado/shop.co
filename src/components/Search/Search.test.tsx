import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { Search } from '.';

describe('<Search />', () => {
  it('should render with default values', () => {
    render(<Search />);

    expect(screen.getByRole('button')).toBeInTheDocument();
    expect(screen.getByRole('button')).toHaveAttribute('type', 'submit');
    expect(screen.getByRole('textbox')).toBeInTheDocument();
    expect(screen.getByRole('textbox')).toHaveAttribute(
      'placeholder',
      'Search for products...',
    );
  });

  it('should render a snapshot', () => {
    const { container } = render(<Search />);

    expect(container).toMatchSnapshot();
  });
});
