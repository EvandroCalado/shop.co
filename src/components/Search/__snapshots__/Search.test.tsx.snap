// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Search /> > should render a snapshot 1`] = `
<div>
  <form
    class="flex w-full items-center gap-2 rounded-full bg-[#f0f0f0] px-4 py-2"
  >
    <button
      type="submit"
    >
      <svg
        class="lucide lucide-search w-4 text-zinc-400"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="11"
          cy="11"
          r="8"
        />
        <path
          d="m21 21-4.3-4.3"
        />
      </svg>
    </button>
    <input
      class="w-full bg-transparent text-sm outline-none"
      placeholder="Search..."
      type="text"
    />
  </form>
</div>
`;
